diff --git a/application/controllers/Appoitments.php b/application/controllers/Appoitments.php
index 73b8b98..0dc1a17 100644
--- a/application/controllers/Appoitments.php
+++ b/application/controllers/Appoitments.php
@@ -69,6 +69,7 @@ class Appoitments extends CI_Controller {
         }
     }
     public function add() {
+	
         if ($this->auth->isLoggedIn()) {
             if ($this->appoitments_model->add()) {
                 $data['success'] = array($this->lang->line("msg_appoitment_added"));
@@ -80,7 +81,7 @@ class Appoitments extends CI_Controller {
         } else redirect('index/login');
     }
     public function update() {
-
+        print_r($_POST);exit;
         if ($this->auth->isLoggedIn()) {
             $data = array();
             $id = $this->input->post('eidt_gf_id');
diff --git a/application/controllers/Branches.php b/application/controllers/Branches.php
index c7eb535..db909b2 100644
--- a/application/controllers/Branches.php
+++ b/application/controllers/Branches.php
@@ -90,6 +90,8 @@ class Branches extends CI_Controller {
             }), array("db" => "city", "dt" => 2, "formatter" => function ($d, $row) {
                 return $this->general_model->getCityName($d);
             }), array("db" => "id", "dt" => 3, "formatter" => function ($d, $row) {
+                return "<a href=\"departments\" id=\"dellink_".$d."\"   data-id=\"$d\" data-toggle=\"tooltip\" title=\"View Department\"><i class=\"glyphicon glyphicon-eye-open\"></i></button>";
+            }), array("db" => "id", "dt" => 4, "formatter" => function ($d, $row) {
                 return "<a href=\"#\" id=\"dellink_".$d."\" class=\"delbtn\"  data-toggle=\"modal\" data-target=\".bs-example-modal-sm\" data-id=\"$d\" data-toggle=\"tooltip\" title=\"Delete\"><i class=\"glyphicon glyphicon-remove\"></i></button>";
             }));
             
diff --git a/application/controllers/Departments.php b/application/controllers/Departments.php
index b7c25a3..9c71329 100644
--- a/application/controllers/Departments.php
+++ b/application/controllers/Departments.php
@@ -95,6 +95,8 @@ class Departments extends CI_Controller {
             $columns = array( array("db" => "department_name", "dt" => 0, "formatter" => function ($d, $row) {
                 return "<a href='#' data-id='$row[id]' class='editbtn' data-toggle='modal' data-target='#edit' data-toggle='tooltip' title='Edit'>".$d."</a>";
             }), array("db" => "id", "dt" => 1, "formatter" => function ($d, $row) {
+                return "<a href=\"wards\" id=\"dellink_".$d."\" title=\"View Wards\"><i class=\"glyphicon glyphicon-eye-open\"></i></button>";
+            }), array("db" => "id", "dt" => 2, "formatter" => function ($d, $row) {
                 return "<a href=\"#\" id=\"dellink_".$d."\" class=\"delbtn\"  data-toggle=\"modal\" data-target=\".bs-example-modal-sm\" data-id=\"$d\" data-toggle=\"tooltip\" title=\"Delete\"><i class=\"glyphicon glyphicon-remove\"></i></button>";
             }));
 
diff --git a/application/controllers/Doctors.php b/application/controllers/Doctors.php
index 8999540..ec0b10d 100644
--- a/application/controllers/Doctors.php
+++ b/application/controllers/Doctors.php
@@ -20,7 +20,10 @@ class Doctors extends CI_Controller {
         $data["breadcrumb"] = array(site_url() =>  $this->lang->line('home'), null =>  $this->lang->line('doctors'));
         if ($this->auth->isLoggedIn() && ($this->auth->isSuperAdmin() || $this->auth->isHospitalAdmin())) {
             $this->load->view('Doctors/index', $data);
-        } 
+        }
+		else if ($this->auth->isLoggedIn() && $this->auth->isPatient()) {
+            $this->load->view('Patient/index', $data);
+        }  
         else if($this->auth->isLoggedIn() && $this->auth->isReceptinest()){
             $this->load->view('Doctors/receptionist', $data);
         }
diff --git a/application/controllers/Hospitals.php b/application/controllers/Hospitals.php
index 69b99d4..330f2f8 100644
--- a/application/controllers/Hospitals.php
+++ b/application/controllers/Hospitals.php
@@ -114,6 +114,8 @@ class Hospitals extends CI_Controller {
             }), array("db" => "city", "dt" => 2, "formatter" => function ($d, $row) {
                 return $this->general_model->getCityName($d);
             }), array("db" => "id", "dt" => 3, "formatter" => function ($d, $row) {
+                return "<a href=\"branches\" id=\"dellink_".$d."\"  title=\"View Branches\"><i class=\"glyphicon glyphicon-eye-open\"></i></button>";
+            }), array("db" => "id", "dt" => 4, "formatter" => function ($d, $row) {
                 return "<a href=\"#\" id=\"dellink_".$d."\" class=\"delbtn\"  data-toggle=\"modal\" data-target=\".bs-example-modal-sm\" data-id=\"$d\" data-toggle=\"tooltip\" title=\"Delete\"><i class=\"glyphicon glyphicon-remove\"></i></button>";
             }));
 
diff --git a/application/controllers/Index.php b/application/controllers/Index.php
index f68e1c5..e2dd558 100644
--- a/application/controllers/Index.php
+++ b/application/controllers/Index.php
@@ -644,7 +644,75 @@ public function privacy_policy(){
 
 	   $this->load->view('index/privacy_policy');
 	
-	}		
+	}
+	
+public function searchDoctor() {
+        if ($this->auth->isLoggedIn()) {
+		
+            $q = $this->input->post("q");
+			$this->load->model('users_model');
+            $result = $this->users_model->searchDoctor($q);
+            //$searcresult = $result;
+			if($result){
+			echo "<ul class='searchlist'>";
+			foreach($result as $Row){
+				echo "<li class='selected-docotr' rel='$Row->FullName' rel1='$Row->id'>$Row->FullName</li>";
+				}
+				echo "</ul>";
+			}else{
+				return false;
+				//echo "<ul class='searchlist'><li class='selected-docotr'>No Doctor Found</li></ul>";
+				}
+        }
+    }
+	
+public function searchBranches() {
+        if ($this->auth->isLoggedIn()) {
+            $q = $this->input->get("q", null, "");
+            $f = $this->input->get("f", null, "");
+            $hid = $this->input->get("hospital_id",null,-1);
+            if($hid == -1 && !$this->auth->isSuperAdmin()){
+                $hid = $this->auth->getHospitalId();
+            }
+            $this->load->model('branches_model');
+            $result = $this->branches_model->search($q,$f,$hid);
+			//print_r($result);exit;
+            echo "<select name='branch_id' class='BranchID allowalphanumeric form-control'>";
+			foreach($result as $Row){
+			echo "<option value='$Row[id]'>$Row[text]</option>";
+			}
+			echo "</select>";
+        }
+    }
+	
+public function searchDepartment() {
+       //error_reporting(0);
+        if ($this->auth->isLoggedIn()) {
+           
+            $result = $this->users_model->DepartmentsByBranchID($this->input->get("branch_id"));
+			//print_r($result);exit;
+            echo "<select name='department_id' class='DepartmentID allowalphanumeric form-control'><option value=''>Please Select</option>";
+			foreach($result as $Row){
+			echo "<option value='$Row->id'>$Row->department_name</option>";
+			}
+			echo "</select>";
+        }
+    }
+	
+public function searchDepartmentDoctor() {
+       //error_reporting(0);
+        if ($this->auth->isLoggedIn()) {
+            $result = $this->users_model->getdoctorsByDepartmentID($this->input->get("dept_id"));
+			if($result){
+			foreach($result as $Row){
+			echo "<input type='checkbox' name='doc_id[]' class='DoctorID' value='$Row->id'>$Row->FullName ";
+			}
+			}else{
+			echo "No Doctors Found.";
+			}
+			
+        }
+    }						
 
 }
 
diff --git a/application/controllers/Nurse.php b/application/controllers/Nurse.php
index 6314710..9791be1 100644
--- a/application/controllers/Nurse.php
+++ b/application/controllers/Nurse.php
@@ -22,6 +22,7 @@ class Nurse extends CI_Controller {
             redirect('index');
         }
         if ($this->auth->isLoggedIn() && ($this->auth->isSuperAdmin() || $this->auth->isHospitalAdmin())) {
+		    $data['Branches'] = $this->branches_model->getHospitalBranches($this->auth->getHospitalId());
             $data['nurses'] = $this->nurse_model->getAllnurse();
             $data["page_title"] = $this->lang->line('nurses');
             $data["breadcrumb"] = array(site_url() => $this->lang->line('home'), null => $this->lang->line('nurses'));
@@ -33,6 +34,14 @@ class Nurse extends CI_Controller {
             $this->load->view('nurse/department');
         }
         else redirect('index/login');
+    }
+	public function wards() {
+        if ($this->auth->isLoggedIn() && ($this->auth->isNurse())) {
+            // $data['bedss'] = $this->beds_model->getAllbeds();
+            $data["page_title"] = $this->lang->line('beds');
+            $data["breadcrumb"] = array(site_url() => $this->lang->line('home'), null => $this->lang->line('beds'));
+            $this->load->view('Nurse/wards', $data);
+        } else redirect('index/login');
     }
     public function beds() {
         if ($this->auth->isLoggedIn() && ($this->auth->isNurse())) {
@@ -169,6 +178,7 @@ class Nurse extends CI_Controller {
                     ->showCheckbox(true)
                     ->from('hms_nurse')
                     ->select('hms_nurse.id as mainid, CONCAT(hms_users.first_name," ",hms_users.last_name) as nursename, hms_hospitals.name as hname, hms_branches.branch_name as bname, hms_departments.department_name as dname,case when hms_nurse.isActive=1 THEN "'.$this->lang->line('active').'" when hms_nurse.isActive=0 THEN "'.$this->lang->line('inactive').'" end as status, hms_nurse.id as action_nurse_id', false)
+					->join('hms_doctors','hms_nurse.doc_id = hms_doctors.id','left')
                     ->join('hms_users','hms_nurse.user_id = hms_users.id','left')
                     ->join('hms_departments','hms_nurse.department_id = hms_departments.id','left')
                     ->join('hms_branches','hms_departments.branch_id = hms_branches.id','left')
diff --git a/application/controllers/Receptionist.php b/application/controllers/Receptionist.php
index 76d3cf0..261a4de 100644
--- a/application/controllers/Receptionist.php
+++ b/application/controllers/Receptionist.php
@@ -17,7 +17,8 @@ class Receptionist extends CI_Controller {
             if($this->auth->isReceptinest()){
                 redirect('index');
             }else if($this->auth->isSuperAdmin() || $this->auth->isHospitalAdmin()){
-                $data['receptionists'] = $this->receptionist_model->getAllreceptionist();
+			    $data['Branches'] = $this->branches_model->getHospitalBranches($this->auth->getHospitalId());
+				$data['receptionists'] = $this->receptionist_model->getAllreceptionist();
                 $data["page_title"] = $this->lang->line('receptionists');
                 $data["breadcrumb"] = array(site_url() => $this->lang->line('home'), null => $this->lang->line('receptionists'));
                 $this->load->view('Receptionist/index', $data);
@@ -41,7 +42,8 @@ class Receptionist extends CI_Controller {
     }
     public function add() {
         if ($this->auth->isLoggedIn() && ($this->auth->isSuperAdmin() || $this->auth->isHospitalAdmin())) {
-            $res = $this->receptionist_model->add();
+            //$res = $this->receptionist_model->add();
+			$res = $this->receptionist_model->addreceptionist();
             $data = $this->auth->parseUserResult($res,$this->lang->line('msg_receptionist_added'));
             $this->session->set_flashdata('data', $data);
             redirect('receptionist/index');
diff --git a/application/controllers/Wards.php b/application/controllers/Wards.php
index 29d1e00..6d063e9 100644
--- a/application/controllers/Wards.php
+++ b/application/controllers/Wards.php
@@ -99,6 +99,8 @@ class Wards extends CI_Controller {
             }),array("db" => "id", "dt" => 2, "formatter" => function ($d, $row) {
                 return $this->beds_model->getTotalAvailableBedsByWard($d);
             }),array("db" => "id", "dt" => 3, "formatter" => function ($d, $row) {
+                return "<a href=\"beds\" id=\"dellink_".$d."\" title=\"View Beds\"><i class=\"glyphicon glyphicon-eye-open\"></i></button>";
+            }),array("db" => "id", "dt" => 4, "formatter" => function ($d, $row) {
                 return "<a href=\"#\" id=\"dellink_".$d."\" class=\"delbtn\"  data-toggle=\"modal\" data-target=\".bs-example-modal-sm\" data-id=\"$d\" data-toggle=\"tooltip\" title=\"Delete\"><i class=\"glyphicon glyphicon-remove\"></i></button>";
             }));
 
@@ -172,4 +174,87 @@ class Wards extends CI_Controller {
             echo json_encode($this->tbl->simple($_GET, $sql_details, $table, $primaryKey, $columns));
         }
     }
+	
+	public function getDTNursewards() {
+        if ($this->auth->isLoggedIn() ) {
+
+            $hid = isset($_GET['hid']) ? $_GET['hid']!="" ? intval($_GET['hid']) : null : null;
+            $bid = isset($_GET['bid']) ? $_GET['bid']!="" ? intval($_GET['bid']) : null : null;
+            $did = isset($_GET['did']) ? $_GET['did']!="" ? intval($_GET['did']) : null : null;
+            $cond = array("hms_beds.isDeleted = 0");
+
+            if($hid == "all")
+                $hid = null;
+            
+            if($did != "all" && $did != null){
+                $wids = $this->wards_model->getWardIdsFromDepartment($did);
+                if(count($wids) == 0){
+                    //If no department created.
+                    //Add dummy id to return nothing
+                    $wids[] = -1;
+                }
+                $ids = implode(",", $wids);
+                $cond[] = "hms_beds.ward_id in (".$ids.")";
+            }else if($bid == "all"){
+                $bids = $this->branches_model->getBracheIds($hid);
+                $ids = $this->wards_model->getWardIdsFromBranch($bids);
+                if(count($ids) == 0){
+                    //If no department created.
+                    //Add dummy id to return nothing
+                    $ids[] = -1;
+                }
+                $ids = implode(",", $ids);
+                $cond[] = "hms_beds.ward_id in (".$ids.")";
+            }else if($bid != null){
+                $ids = $this->wards_model->getWardIdsFromBranch($bid);
+                if(count($ids) == 0){
+                    //If no department created.
+                    //Add dummy id to return nothing
+                    $ids[] = -1;
+                }
+                $ids = implode(",", $ids);
+                $cond[] = "hms_beds.ward_id in (".$ids.")";
+            }else if($hid!=null){
+                $ids = $this->wards_model->getWardIdsFromHospital($hid);
+                if(count($ids) == 0){
+                    //If no department created.
+                    //Add dummy id to return nothing
+                    $ids[] = -1;
+                }
+                $ids = implode(",", $ids);
+                $cond[] = "hms_beds.ward_id in (".$ids.")";
+            }else{
+                $hids = $this->hospitals_model->getHospicalIds();
+                $ids = $this->wards_model->getWardIdsFromHospital($hids);
+                if(count($ids) == 0){
+                    //If no department created.
+                    //Add dummy id to return nothing
+                    $ids[] = -1;
+                }
+                $ids = implode(",",$ids);
+                $cond[] = "hms_beds.ward_id in (".$ids.")";
+            }
+
+            //New Library
+            $this->load->library('datatables');		
+            $this->datatables
+                ->showIndex(true)
+                ->from('hms_beds')
+                ->select('hms_beds.id as mainid,hms_wards.ward_name as wname, hms_beds.bed as bed, case when hms_beds.isAvailable=0 then "No" when hms_beds.isAvailable=1 then "Yes" end as status, CONCAT(hms_users.first_name," ",hms_users.last_name) as pname, hms_inpatient.id as action_pid',false)
+                ->join('hms_wards','hms_beds.ward_id = hms_wards.id','left')
+                ->join('hms_inpatient','hms_beds.id = hms_inpatient.bed_id','left')
+                ->join('hms_users','hms_inpatient.user_id = hms_users.id','left')
+                ->edit_column('pname','<a href="'.site_url().'nurse/inpatient?sip=1&pid=$1">$2</a>','action_pid, pname')
+                ->unset_column('action_pid');
+            
+            //Set condition to new library
+            foreach($cond as $con){
+                $this->datatables->where($con);
+            }
+
+            //Call new library for output
+            echo $this->datatables->generate('json');
+
+        }
+    }
 }
diff --git a/application/language/english/menu_lang.php b/application/language/english/menu_lang.php
index 4d3befc..dfddc18 100644
--- a/application/language/english/menu_lang.php
+++ b/application/language/english/menu_lang.php
@@ -28,12 +28,12 @@ $lang['changePassword'] = "Change Password";
 $lang['appoitments'] = "Appoitments";
 $lang['main_healthinsuranceprovider'] = "Health Insurance Provider";
 $lang['main_inpatient_history'] = "In-Patient History";
-$lang['main_inpatient'] = "In-Patient";
+$lang['main_inpatient'] = "In-Patients";
 $lang['main_hospital_admin'] = 'Hospital Admins';
 $lang['main_availability'] = 'Availability';
 $lang['main_other'] = 'Others';
 $lang['main_about'] = 'About';
-$lang['reports'] = 'Test Reports';
+$lang['reports'] = 'Reports';
 $lang['orders'] = 'Orders';
 $lang['main_patient_report'] = 'Patient Report';
 $lang['main_appoitment_report'] = 'Appoitment Report';
diff --git a/application/models/Branches_model.php b/application/models/Branches_model.php
index 9954af5..4c7b2a5 100644
--- a/application/models/Branches_model.php
+++ b/application/models/Branches_model.php
@@ -193,4 +193,25 @@ class Branches_model extends CI_Model {
         }
         return $ids;
     }
+	
+function getHospitalBranches($hospital_id) {
+        if($this->auth->isDoctor() || $this->auth->isSuperAdmin() || $this->auth->isHospitalAdmin() || $this->auth->isReceptinest()){
+            $bids = $this->getBranchesIds();
+            $this->db->where_in("id",$bids);       
+        }
+
+        if($hospital_id > 0){
+            $this->db->where('hospital_id',$hospital_id);
+        }else{
+            $hids = $this->auth->getAllHospitalIds();
+            if(count($hids) == 0){ $hids[] = -1;} 
+            $this->db->where_in('hospital_id',$hids);
+        }
+
+        $this->db->where("isDeleted",0);
+        $this->db->where("isActive",1);
+        $this->db->select("id,CONCAT(`branch_name`) as text", false);
+        $res = $this->db->get($this->tblname);
+        return $res->result();
+    }	
 }
diff --git a/application/models/Doctors_model.php b/application/models/Doctors_model.php
index 03147f8..fadbcf9 100644
--- a/application/models/Doctors_model.php
+++ b/application/models/Doctors_model.php
@@ -1194,8 +1194,7 @@ class Doctors_model extends CI_Model {
 	
 	public function validateMobileNumber($mobnumber){
 		
-		
-		$checkmob = $this->db->query("SELECT `id` FROM `hms_users` WHERE `mobile` = '$mobnumber'")->row();
+		$checkmob = $this->db->query("SELECT `id` FROM `hms_users` WHERE `mobile` = '$mobnumber' AND `isRegister` !='0'")->row();
 		if(!empty($checkmob)){
 			return  array('Status' => 1);
 		}
@@ -1205,9 +1204,8 @@ class Doctors_model extends CI_Model {
 	}
 	
 	public function validateEmailAvailable($emailval){
-		
-		
-		$checkemail = $this->db->query("SELECT `id` FROM `hms_users` WHERE `useremail` = '$emailval'")->row();
+				
+		$checkemail = $this->db->query("SELECT `id` FROM `hms_users` WHERE `useremail` = '$emailval'  AND `isRegister` !='0'")->row();
 		if(!empty($checkemail)){
 			return  array('Status' => 1);
 		}
diff --git a/application/models/Nurse_model.php b/application/models/Nurse_model.php
index 79fc4d1..f407707 100644
--- a/application/models/Nurse_model.php
+++ b/application/models/Nurse_model.php
@@ -101,7 +101,11 @@ class Nurse_model extends CI_Model {
             $nurse['created_at'] = date("Y-m-d H:i:s");
             if(isset($data['isActive']))
                 $nurse['isActive'] = intval($data['isActive']);
-            if ($this->db->insert($this->tblname, $nurse)) {
+				$doctorsid = $this->input->post('doc_id') ? $this->input->post('doc_id') : 0;
+            if($doctorsid){
+            foreach ($doctorsid as $did) {
+			    $nurse['doc_id'] = $did;
+				$this->db->insert($this->tblname, $nurse);
 				$id = $this->db->insert_id();
 				$this->logger->log("New nurse added", Logger::Nurse, $id);
 				
@@ -127,6 +131,7 @@ class Nurse_model extends CI_Model {
                         }   
                     }
                 }
+				}
                 return true;
             } else {
                 return false;
diff --git a/application/models/Receptionist_model.php b/application/models/Receptionist_model.php
index 6f6c35a..50482d1 100644
--- a/application/models/Receptionist_model.php
+++ b/application/models/Receptionist_model.php
@@ -153,6 +153,72 @@ class Receptionist_model extends CI_Model {
         }
         return true;
     }
+	
+	function addreceptionist() {
+        $data = $_POST;
+        $data['role'] = $this->auth->getReceptienstRoleType();
+        $rec_id = $this->auth->addUser($data);
+		//print_r($this->input->post('doc_id'));exit;
+        
+        if($rec_id === false){
+            return false;
+        }else if($rec_id < 0){
+            return $rec_id;
+        }
+        else{
+            $rec = array();
+            $rec['user_id'] = $rec_id;
+            //$rec['doc_id'] = isset($data['doc_id']) ? $data['doc_id'] : 0;
+            if(isset($data['qualification']))
+                $rec['qualification'] = $data['qualification'];
+            if(isset($data['experience']))
+                $rec['experience'] = $data['experience'];
+            $rec['created_at'] = date("Y-m-d H:i:s");
+            if(isset($data['isActive']))
+                $rec['isActive'] = intval($data['isActive']);
+				$doctorsid = $this->input->post('doc_id') ? $this->input->post('doc_id') : 0;
+				if($doctorsid){
+            foreach ($doctorsid as $did) {
+			    $rec['doc_id'] = $did;
+			    $this->db->insert($this->tblname, $rec);
+				$id = $this->db->insert_id();
+                $this->logger->log("New receptionist added", Logger::Receptionist, $id);
+                
+				if(isset($did) && $did != ""){
+                    //find doctor user_id which is linked with this receptionist
+                    $this->db->where('id', $did);
+                    $doctor = $this->db->get('hms_doctors')->row_array();
+                    //find doctor name from user table
+                    $this->db->where('id', $doctor['user_id']);
+                    $dname = $this->db->get('hms_users')->row_array();
+
+                    //sent notification to doctor
+                    $this->notification->saveNotification($doctor['user_id'],"New receptionist <b>".$data['first_name']." ".$data['last_name']."</b> is linked with you");
+
+                    if(isset($rec['user_id']) && $rec['user_id'] != ""){
+                        //sent notification to receptionist
+                        $this->notification->saveNotification($rec['user_id'], "You are linked with <b>".$dname['first_name']." ".$dname['last_name']."</b> doctor as Receptionist");              
+                    }
+
+                    if($this->auth->isSuperAdmin()){
+                        if(isset($data['hospital_id']) && $data['hospital_id'] != ""){
+                            //find hospital admin
+                            $this->db->where('hospital_id', $data['hospital_id']);
+                            $hadmin = $this->db->get('hms_hospital_admin')->row_array();
+                            //sent notification to hospital admin
+                            $this->notification->saveNotification($hadmin['user_id'], "New receptionist <b>".$data['first_name']." ".$data['last_name']."</b> is linked with doctor: <b>".$dname['first_name']." ".$dname['last_name']."</b>");
+                        }
+                    }
+                }
+                
+            }
+			return true;
+			 } else {
+                return false;
+            }
+        }
+        return true;
+    }
     
     function update($id) {
         $data = $_POST;
diff --git a/application/models/Users_model.php b/application/models/Users_model.php
index 0d30441..4f45e06 100644
--- a/application/models/Users_model.php
+++ b/application/models/Users_model.php
@@ -390,7 +390,7 @@ class Users_model extends CI_Model {
             $mail_data['body'] = 'Welcome to MyPulse,<br>'.$username.' has register you as '.$role.'<br>To complete your MyPulse profile. Please verify your account by click on following link and update your password <br> <a href="'.$url.'">Verify Account</a>';
             $mail_data['subject'] = 'MyPulse Registration';
             $mail_data['email'] = $data['useremail'];
-            $this->sendmail->send($mail_data);
+            //$this->sendmail->send($mail_data);
 			$mobno = $data['mobile'];
             $SMSText = ' Welcome to MyPulse. '.$username.' has register you as '.$role.' To complete your MyPulse profile. Please verify your email '.$email.'';
 $json = json_decode(file_get_contents("https://smsapi.engineeringtgr.com/send/?Mobile=9739195391&Password=mypulse123&Message=".urlencode($SMSText)."&To=".urlencode($mobno)."&Key=raisiVvTbgKISshQjnMNGr"),true);
@@ -1076,6 +1076,44 @@ public function updateStaffPassword()
                 return false;   
         }           
         return false;
-    }		
+    }
+	
+public function searchDoctor($q = NULL){
+       
+	   $search = explode(' ',$q);
+		$ocusearch = count($search);
+		if($ocusearch > '1'){
+		
+$Result = $this->db->query("SELECT doc.`id`,CONCAT_WS(' ',users.`first_name`,users.`MiddleName`,users.`last_name`) AS FullName FROM `hms_doctors` AS doc INNER JOIN `hms_users` AS users ON users.`id`=doc.user_id INNER JOIN `hms_hospitals` AS hos ON hos.`id` = doc.`user_id` WHERE users.`first_name` LIKE '%".$search[0]."%' OR users.`MiddleName` LIKE '%".$search[1]."%' OR users.`last_name` LIKE '%".$search[1]."%' AND users.isDeleted='0' AND users.isActive='1' AND doc.isDeleted='0' AND doc.isActive='1' AND hos.id = '".$this->session->userdata('hospital_id')."' AND users.role='3' GROUP BY doc.`user_id` ");
+		}else{
+			
+			$Result = $this->db->query("SELECT doc.`id`,CONCAT_WS(' ',users.`first_name`,users.`MiddleName`,users.`last_name`) AS FullName FROM `hms_doctors` AS doc INNER JOIN `hms_users` AS users ON users.`id`=doc.user_id INNER JOIN `hms_hospitals` AS hos ON hos.`id` = doc.`user_id` WHERE users.`first_name` LIKE '%".$search[0]."%' OR users.`MiddleName` LIKE '%".$search[0]."%' OR users.`last_name` LIKE '%".$search[0]."%' AND users.isDeleted='0' AND users.isActive='1' AND doc.isDeleted='0' AND doc.isActive='1' AND hos.id = '".$this->session->userdata('hospital_id')."' AND users.role='3' GROUP BY doc.`user_id` ");
+			}
+		if($Result->num_rows() > 0){
+		    return $Result->result();
+		}else{
+		  return false;
+		}	
+	}
+	
+public function DepartmentsByBranchID($BranchID = NULL){
+
+    $Result =  $this->db->query("SELECT `id`,`department_name` FROM `hms_departments` WHERE `isActive`='1' AND `isDeleted`='0' AND `branch_id`='$BranchID'")->result();
+    if($Result){
+		return $Result;	  
+	   }else{
+	    return false; 
+	   }
+    }				
   	
+public function getdoctorsByDepartmentID($DeptID = NULL){
+
+    $Result =  $this->db->query("SELECT doc.`id`,doc.`user_id`,CONCAT_WS(' ',usr.`first_name`,usr.`MiddleName`,usr.`last_name`) AS FullName FROM `hms_doctors` AS doc INNER JOIN `hms_users` AS usr ON usr.`id`=doc.`user_id` WHERE usr.`isActive`='1' AND usr.`isDeleted`='0' AND doc.`department_id`='$DeptID' GROUP BY doc.user_id ")->result();
+    if($Result){
+		return $Result;	  
+	   }else{
+	    return false; 
+	   }
+    }	
+	
 }
diff --git a/application/views/Appoitments/doctor.php b/application/views/Appoitments/doctor.php
index f010b89..ad507da 100644
--- a/application/views/Appoitments/doctor.php
+++ b/application/views/Appoitments/doctor.php
@@ -104,40 +104,40 @@ $this->load->view("template/left.php");
 									<select name="user_id" id="user_id" class="form-control" style="width: 100%;">
 									</select>
 								</div>
+								<input type="hidden" name="doctor_id" id="doctor_id" value="<?php echo $this->auth->getDoctorId();?>" />
 								<div class="form-group col-md-6">
+									<label><?php echo $this->lang->line('labels')['doctorAvailability'];?></label>
+									<br>
+									<span id="docAvailability"></span>
+								</div>
+								<div class="form-group col-md-6 hide">
                                     <label><?php echo $this->lang->line('labels')['selectHospital'];?></label>
                                     <select id="hospital_id" class=" form-control" style="width: 100%">
 					                </select>
                                 </div>
 							</div>
 							<div class="col-md-12">
-								<div class="form-group col-md-6">
+								<div class="form-group col-md-6 hide">
                                     <label><?php echo $this->lang->line('labels')['selectBranch'];?></label>
                                     <select id="branch_id" class=" form-control" style="width: 100%">
 					                </select>
                                 </div>
-								<div class="form-group col-md-6">
+								<div class="form-group col-md-6 hide">
                                     <label><?php echo $this->lang->line('labels')['selectDepartment'];?></label>
                                     <select id="department_id" name="department_id" class=" form-control" style="width: 100%">
 					                </select>
                                 </div>
-								<input type="hidden" name="doctor_id" id="doctor_id" value="<?php echo $this->auth->getDoctorId();?>" />
+								
 							</div>
-							<div class="col-md-12">
-								<div class="form-group col-md-12">
-									<label><?php echo $this->lang->line('labels')['doctorAvailability'];?></label>
-									<br>
-									<span id="docAvailability"></span>
-								</div>
-							</div>	
+								
 							<div class="col-md-12">
 								<div class="form-group col-md-6">
 									<label><?php echo $this->lang->line('labels')['appoitment_date'];?></label>
-									<input class="form-control date-picker-nopast" type="text" placeholder="<?php echo $this->lang->line('labels')['appoitment_date'];?>" name="appoitment_date" id="appoitment_date" />
+									<input class="form-control allowalphanumeric  date-picker-nopast" type="text" placeholder="<?php echo $this->lang->line('labels')['appoitment_date'];?>" name="appoitment_date" id="appoitment_date" />
 								</div>
 								<div class="form-group col-md-6">
 									<label><?php echo $this->lang->line('labels')['appoitment_sloat'];?></label>
-									<select class="form-control" type="text" name="appoitment_sloat" id="appoitment_sloat">
+									<select class="form-control allowalphanumeric" type="text" name="appoitment_sloat" id="appoitment_sloat">
 									</select>
 									<br>
 									<span id="noApptTimeSloat" style='color:#BC4442;display:none'><?php echo $this->lang->line('labels')['noApptTimeSloat'];?></span>
@@ -146,11 +146,11 @@ $this->load->view("template/left.php");
 							<div class="col-md-12">
 								<div class="form-group col-md-6">
 									<label><?php echo $this->lang->line('labels')['appoitment_reason'];?></label>
-									<textarea class="form-control " type="text" placeholder="<?php echo $this->lang->line('labels')['appoitment_reason'];?>" name="reason" id="reason" rows="3"></textarea>
+									<textarea class="form-control allowalphanumeric " type="text" placeholder="<?php echo $this->lang->line('labels')['appoitment_reason'];?>" name="reason" id="reason" rows="3"></textarea>
 								</div>
 								<div class="form-group col-md-6">
 									<label><?php echo $this->lang->line('labels')['remark'];?></label>
-									<textarea  class="form-control " type="text" placeholder="<?php echo $this->lang->line('labels')['remark'];?>" name="remarks" id="remarks" rows="3"></textarea>
+									<textarea  class="form-control allowalphanumeric " type="text" placeholder="<?php echo $this->lang->line('labels')['remark'];?>" name="remarks" id="remarks" rows="3"></textarea>
 								</div>
 							</div>				  		
 						</div>
@@ -385,7 +385,7 @@ $this->load->view("template/footer.php");
 			$selectize_hospital_id[0].selectize.clear();
 			$selectize_user_id[0].selectize.enable();
 			$selectize_user_id[0].selectize.clear();
-			$("#appoitment_date").attr('disabled',true);
+			//$("#appoitment_date").attr('disabled',true);
 			$("#appoitment_sloat").attr('disabled',true);
 			$("#appoitment_sloat").html('');
 			$("#reason").attr('disabled',false);
@@ -709,7 +709,7 @@ $this->load->view("template/footer.php");
 			labelField: "text",
 			searchField: "text",
 			loadThrottle: 500,
-			placeholder: "Enter user Email or Mobile number or Aadhaar number",
+			placeholder: "Enter User Email or Mobile number",
 			preload:true,
 			create: function(input,callback){
 				userCreateCallBack = callback;
@@ -720,7 +720,7 @@ $this->load->view("template/footer.php");
 			},
 			render: {
 				option_create: function(data, escape) {
-					return '<div class="create"><strong><?=$this->lang->line('unregUser');?></strong></div>';
+					return '<div class="create"><strong style="color:red;"><?=$this->lang->line('unregUser');?></strong></div>';
 				},
 				option: function(item, escape) {
 					console.log(item);
diff --git a/application/views/Appoitments/receptionist.php b/application/views/Appoitments/receptionist.php
index 6d6da51..5aa4270 100644
--- a/application/views/Appoitments/receptionist.php
+++ b/application/views/Appoitments/receptionist.php
@@ -1,3 +1,9 @@
+<style>
+#country-list{float:left;list-style:none;margin-top:-3px;padding:0;width:190px;position: absolute;/*overflow: scroll; height:150px;*/}
+#country-list li{padding: 10px; background: #f0f0f0; border-bottom: #bbb9b9 1px solid;}
+#country-list li:hover{background:#ece3d2;cursor: pointer;}
+#search-box{padding: 10px;border: #a8d4b1 1px solid;border-radius:4px;}
+</style>
 <?php
 /**
  * @author Yogesh Patel
@@ -111,18 +117,23 @@ $this->load->view("template/left.php");
 									</select>
 								</div>
 								<div class="form-group col-md-6">
+									<label>Search Doctor Name</label>
+									<input type="text" placeholder="Enter Doctor Name" name="DoctorName" class="DoctorName form-control allowalphanumeric" value=""  />                             <input type="hidden" name="doctor_id" id="DoctorID" class="DoctorID" value=""  />
+                                    <div id="suggesstion-box"></div>
+								</div>
+								<div class="form-group col-md-6 ">
                                     <label><?php echo $this->lang->line('labels')['selectHospital'];?></label>
                                     <select id="hospital_id" class=" form-control allowalphanumeric" >
 					                </select>
                                 </div>
 							</div>
 							<div class="col-md-12">
-								<div class="form-group col-md-6">
+								<div class="form-group col-md-6 hide">
                                     <label><?php echo $this->lang->line('labels')['selectBranch'];?></label>
                                     <select id="branch_id" class=" form-control allowalphanumeric" >
 					                </select>
                                 </div>
-								<div class="form-group col-md-6">
+								<div class="form-group col-md-6 hide">
                                     <label><?php echo $this->lang->line('labels')['selectDepartment'];?></label>
                                     <select id="department_id" name="department_id" class=" form-control allowalphanumeric" >
 					                </select>
@@ -664,7 +675,7 @@ $this->load->view("template/footer.php");
 			labelField: "text",
 			searchField: "text",
 			loadThrottle: 500,
-			placeholder: "Enter user id",
+			placeholder: "Enter User Email or Mobile number ",
 			preload:true,
 			create: function(input,callback){
 				userCreateCallBack = callback;
@@ -747,7 +758,7 @@ $this->load->view("template/footer.php");
 			}
 		});
 
-		var $selectize_department_id = $("#department_id").selectize({
+		<?php /*?>var $selectize_department_id = $("#department_id").selectize({
 			valueField: "id",
 			labelField: "text",
 			searchField: "text",
@@ -918,7 +929,7 @@ $this->load->view("template/footer.php");
 					})
 				});                 
 			}
-		});
+		});<?php */?>
 
 			
 		var $selectize_doctor_id1 = $("#doctor_id1").selectize({
@@ -1068,6 +1079,42 @@ $this->load->view("template/footer.php");
 		
 		$("#sel_date").val("");	
 		loadTable("all","all");	
+		
+		$('.DoctorName').on('keyup', function(){
+		   $SearchTerm = $(this).val();
+		   if($SearchTerm.length > 3){
+		   $.ajax({
+					url: "<?php echo site_url(); ?>/index/searchDoctor/",
+					type: "POST",
+					data: {"q":$SearchTerm},
+					error: function() {
+						callback();
+					},
+					success: function(res) {
+						//res = $.parseJSON(res);
+						/*$.each($.parseJSON(res), function(k, v) {
+   						 //alert(k['id'] + ' is ' + v['id']);
+						});*/
+						if(res){
+						$("#suggesstion-box").show();
+			$("#suggesstion-box").html(res);
+			$(".DoctorName").css("background","#FFF");
+						}else{
+							$(".DoctorID").val('');
+							}
+					}
+				});
+		   }
+		});
+		$('body').delegate('.selected-docotr','click',function(){
+			//alert($(this).attr('rel'));
+			selectDoctor($(this).attr('rel'),$(this).attr('rel1'));
+			});
+		function selectDoctor(DName,DID) {
+		$(".DoctorName").val(DName);
+		$(".DoctorID").val(DID);
+		$("#suggesstion-box").hide();
+		}
 	});
 
 </script>
\ No newline at end of file
diff --git a/application/views/Branches/index.php b/application/views/Branches/index.php
index 8b3888c..8edc16b 100644
--- a/application/views/Branches/index.php
+++ b/application/views/Branches/index.php
@@ -38,7 +38,7 @@ $this->load->view("template/left.php");
                             <div class="col-md-12">
 								<table id="branches" class="table table-striped table-bordered table-hover table-checkable order-column valign-middle">
 									<thead>
-										<tr><th style="width:10px"></th><th><?php echo $this->lang->line('tableHeaders')['branch'];?></th><th><?php echo $this->lang->line('tableHeaders')['phoneNumber'];?></th><th><?php echo $this->lang->line('tableHeaders')['city'];?></th><th width="20px">#</th>
+										<tr><th style="width:10px"></th><th><?php echo $this->lang->line('tableHeaders')['branch'];?></th><th><?php echo $this->lang->line('tableHeaders')['phoneNumber'];?></th><th><?php echo $this->lang->line('tableHeaders')['city'];?></th><th><?php echo $this->lang->line('tableHeaders')['department'];?></th><th width="20px">#</th>
 										</tr>
 									</thead>											
 									<tbody>
diff --git a/application/views/Departments/index.php b/application/views/Departments/index.php
index 11183db..1547864 100644
--- a/application/views/Departments/index.php
+++ b/application/views/Departments/index.php
@@ -46,7 +46,7 @@ $this->load->view("template/left.php");
 								
 								<table id="departments" class="table table-striped table-bordered table-hover table-checkable order-column valign-middle">
 									<thead>
-										<tr><th style="width:10px"></th><th><?php echo $this->lang->line('tableHeaders')['department'];?></th><th width="20px">#</th>
+										<tr><th style="width:10px"></th><th><?php echo $this->lang->line('tableHeaders')['department'];?></th><th><?php echo $this->lang->line('tableHeaders')['ward'];?></th><th width="20px">#</th>
 										</tr>
 									</thead>
 									
diff --git a/application/views/Hospitals/index.php b/application/views/Hospitals/index.php
index 1e4f316..ed2f933 100644
--- a/application/views/Hospitals/index.php
+++ b/application/views/Hospitals/index.php
@@ -24,7 +24,7 @@ $this->load->view("template/left.php");
 	                       
 							<table id="hospitals" class="table table-striped table-bordered table-hover table-checkable order-column valign-middle">
 								<thead>
-									<tr><th style="width:10px"></th><th><?php echo $this->lang->line('tableHeaders')['name'];?></th><th><?php echo $this->lang->line('tableHeaders')['licenseStatus'];?></th><th><?php echo $this->lang->line('tableHeaders')['city'];?></th><th  width="20px">#</th>
+									<tr><th style="width:10px"></th><th><?php echo $this->lang->line('tableHeaders')['name'];?></th><th><?php echo $this->lang->line('tableHeaders')['licenseStatus'];?></th><th><?php echo $this->lang->line('tableHeaders')['city'];?></th><th><?php echo $this->lang->line('tableHeaders')['branch'];?></th><th  width="20px">#</th>
 									</tr>
 								</thead>
 								<tbody>
diff --git a/application/views/Nurse/index.php b/application/views/Nurse/index.php
index b01c410..f614643 100644
--- a/application/views/Nurse/index.php
+++ b/application/views/Nurse/index.php
@@ -46,7 +46,7 @@ $this->load->view("template/left.php");
 
 	    <div class="modal fade" id="edit" tabindex="-1" role="dialog" aria-labelledby="edit" aria-hidden="true">
 			<div class="modal-dialog modal-lg">
-				<form action="<?php echo site_url(); ?>/nurse/update" method="post" id="form" enctype="multipart/form-data">
+				<form autocomplete="off" action="<?php echo site_url(); ?>/nurse/update" method="post" id="form" enctype="multipart/form-data">
 				<input type="hidden" name="eidt_gf_id" id="eidt_gf_id">
 				<div class="modal-content">
 				  	<div class="modal-header">
@@ -112,24 +112,49 @@ $this->load->view("template/left.php");
   											<div class="panel-body"><?php echo $this->lang->line('labels')['hospitalAssociation'];?></div>
 										</div>
 										<div class="col-md-12">
-										    <div class="form-group col-md-6">
+										    <div class="form-group col-md-6 hide">
 												<label><?php echo $this->lang->line('labels')['selectHospital'];?>*</label>
 												<select name="hospital_id"  id="hospital_id" class=" form-control textinputfields" style="width: 77%">
 												<option value="">Please Select</option>
                                                 </select>
 											</div>
-											<div class="form-group col-md-6">
+											<div class="form-group col-md-6 hide">
 												<label><?php echo $this->lang->line('labels')['selectBranch'];?>*</label>
-												<select name="branch_id" id="branch_id" class=" form-control textinputfields" style="width: 77%">
+												<select name="branch_idss" id="branch_id" class=" form-control textinputfields" style="width: 77%">
 												<option value="">Please Select</option>
                                                 </select>
 											</div>
+											<div class="form-group col-md-6 " >
+												<label><?php echo $this->lang->line('labels')['selectBranch'];?>*</label>
+												<select name="branch_id" class="branch_id form-control allowalphanumeric" style="width: 77%" >
+												<option value="">Please Select</option>
+												<?php 
+												foreach($Branches as $br){
+												?>
+												<option value="<?php echo $br->id; ?>"><?php echo $br->text; ?></option>
+												<?php } ?>
+												</select>
+											</div>
+											<div class="form-group col-md-6 ">
+												<label><?php echo $this->lang->line('labels')['selectDepartment'];?>*</label>
+												<div class="ddepartment-list">
+												<select name="department_id"  class="DepartmentID form-control allowalphanumeric department_id" style="width: 77%">
+												<option value="">Please Select</option>
+                                                </select>
+												</div>
+											</div>
 											
 										</div>
 										<div class="col-md-12">
-											<div class="form-group col-md-6">
+											<label><?php echo $this->lang->line('labels')['selectDoctor'];?></label>
+											<div class="doctors-list">
+											
+											</div>
+										</div>
+										<div class="col-md-12">
+											<div class="form-group col-md-6 hide">
 												<label><?php echo $this->lang->line('labels')['selectDepartment'];?>*</label>
-												<select name="department_id" id="department_id" class=" form-control textinputfields" style="width: 77%">
+												<select name="department_idss" id="department_id" class=" form-control textinputfields" style="width: 77%">
 												<option value="">Please Select</option>
                                                 </select>
 											</div>
@@ -633,6 +658,44 @@ $this->load->view("template/footer.php");
 				    }
 				});
 
+
+$('.branch_id').on('change',function(){
+   $HospitalID = '<?php echo $this->session->userdata('hospital_id'); ?>';
+   $branchid = $(this).val();
+   //$(".department_id").attr("disabled",false);
+   $.ajax({
+				url: "<?php echo site_url(); ?>index/searchDepartment/",
+				type: "GET",
+				data: { "hospital_id":$HospitalID,"branch_id":$branchid,"f":"department_name"},
+				success: function(results) {
+				$(".ddepartment-list").html(results);
+					
+				},
+				error: function() {
+					callback();
+				}
+			})
+  
+  });
+    
+$(document).on('change','.DepartmentID',function(){
+
+   $departmentid = $(this).val();
+   //$(".department_id").attr("disabled",false);
+   $.ajax({
+				url: "<?php echo site_url(); ?>index/searchDepartmentDoctor/",
+				type: "GET",
+				data: {"dept_id":$departmentid},
+				success: function(results) {
+				$(".doctors-list").html(results);
+					
+				},
+				error: function() {
+					callback();
+				}
+			})
+
+}); 
 					
 
 			});
diff --git a/application/views/Receptionist/dashboard.php b/application/views/Receptionist/dashboard.php
index c2de31b..556183d 100644
--- a/application/views/Receptionist/dashboard.php
+++ b/application/views/Receptionist/dashboard.php
@@ -228,12 +228,12 @@
                             </div>
                         </div>
                         <div class="col-md-12">
-                            <div class="form-group col-md-6">
+                            <div class="form-group col-md-6 hide">
                                 <label><?php echo $this->lang->line('labels')['selectBranch'];?></label>
                                 <select id="branch_id" class=" form-control" style="width: 100%">
                                 </select>
                             </div>
-                            <div class="form-group col-md-6">
+                            <div class="form-group col-md-6 hide">
                                 <label><?php echo $this->lang->line('labels')['selectDepartment'];?></label>
                                 <select id="department_id" name="department_id" class=" form-control" style="width: 100%">
                                 </select>
@@ -243,22 +243,21 @@
                                 <select name="doctor_id" id="doctor_id" class=" form-control" style="width: 100%">
                                 </select>
                             </div>
-                        </div>
-                        <div class="col-md-12">
-                            <div class="form-group col-md-12">
+							<div class="form-group col-md-6">
                                 <label><?php echo $this->lang->line('labels')['doctorAvailability'];?></label>
                                 <br>
                                 <span id="docAvailability"></span>
                             </div>
                         </div>
+                        
                         <div class="col-md-12">
                             <div class="form-group col-md-6">
                                 <label><?php echo $this->lang->line('labels')['appoitment_date'];?></label>
-                                <input class="form-control date-picker-nopast" type="text" placeholder="<?php echo $this->lang->line('labels')['appoitment_date'];?>" name="appoitment_date" id="appoitment_date" />
+                                <input class="form-control date-picker-nopast allowalphanumeric" type="text" placeholder="<?php echo $this->lang->line('labels')['appoitment_date'];?>" name="appoitment_date" id="appoitment_date" />
                             </div>
                             <div class="form-group col-md-6">
                                 <label><?php echo $this->lang->line('labels')['appoitment_sloat'];?></label>
-                                <select class="form-control" type="text" name="appoitment_sloat" id="appoitment_sloat">
+                                <select class="form-control allowalphanumeric" type="text" name="appoitment_sloat" id="appoitment_sloat">
                                 </select>
                                 <span id="noApptTimeSloat" style='color:#BC4442;display:none'><?php echo $this->lang->line('labels')['noApptTimeSloat'];?></span>
                             </div>
@@ -266,11 +265,11 @@
                         <div class="col-md-12">	
                             <div class="form-group col-md-6">
                                 <label><?php echo $this->lang->line('labels')['appoitment_reason'];?></label>
-                                <textarea class="form-control " type="text" placeholder="<?php echo $this->lang->line('labels')['appoitment_reason'];?>" name="reason" id="reason" rows="3"></textarea>
+                                <textarea class="form-control allowalphanumeric " type="text" placeholder="<?php echo $this->lang->line('labels')['appoitment_reason'];?>" name="reason" id="reason" rows="3"></textarea>
                             </div>
                             <div class="form-group col-md-6">
                                 <label><?php echo $this->lang->line('labels')['remark'];?></label>
-                                <textarea  class="form-control " type="text" placeholder="<?php echo $this->lang->line('labels')['remark'];?>" name="remarks" id="remarks" rows="3"></textarea>
+                                <textarea  class="form-control allowalphanumeric " type="text" placeholder="<?php echo $this->lang->line('labels')['remark'];?>" name="remarks" id="remarks" rows="3"></textarea>
                             </div>
                         </div>				  		
                     </div>
@@ -771,7 +770,7 @@
 			labelField: "text",
 			searchField: "text",
 			loadThrottle: 500,
-			placeholder: "Enter user id",
+			placeholder: "Enter user Email or Mobile number",
 			preload:true,
 			create: function(input,callback){
 				userCreateCallBack = callback;
@@ -782,7 +781,7 @@
 			},
 			render: {
 				option_create: function(data, escape) {
-					return '<div class="create"><strong><?=$this->lang->line('unregUser');?></strong></div>';
+					return '<div class="create"><strong style="color:red;"><?=$this->lang->line('unregUser');?></strong></div>';
 				},
 				option: function(item, escape) {
 					console.log(item);
diff --git a/application/views/Receptionist/index.php b/application/views/Receptionist/index.php
index a5971b5..599cffb 100644
--- a/application/views/Receptionist/index.php
+++ b/application/views/Receptionist/index.php
@@ -14,7 +14,7 @@ $this->load->view("template/left.php");
 						<div class="card-head">
 							<header><?php echo $this->lang->line('receptionists');?></header>
 							<div class="custome_card_header">
-								<a class="btn btn-success m-b-sm addbtn" data-toggle="tooltip" href="javascript:void(0);" data-toggle="modal" data-target="#edit" style=""><?php echo $this->lang->line('buttons')['addNew'];?></a>
+								<a class="btn btn-success m-b-sm addbtn getbranches" data-toggle="tooltip" href="javascript:void(0);" data-toggle="modal" data-target="#edit" style=""><?php echo $this->lang->line('buttons')['addNew'];?></a>
 								<a class="btn btn-danger m-b-sm multiDeleteBtn" data-at="receptionist" href="javascript:void(0);"  style="margin-left:10px"><?php echo $this->lang->line('buttons')['delete'];?></a>
 								<?php $this->load->view('template/exbtn');?>
 							</div>
@@ -110,35 +110,63 @@ $this->load->view("template/left.php");
   											<div class="panel-body"><?php echo $this->lang->line('labels')['hospitalAssociation'];?></div>
 										</div>
 										<div class="col-md-12">
-										    <div class="form-group col-md-6">
+										    <div class="form-group col-md-6 hide">
 												<label><?php echo $this->lang->line('labels')['selectHospital'];?>*</label>
 												<select name="hospital_id"  id="hospital_id" class=" form-control textinputfields" style="width: 77%">
 												<option value="">Please Select</option>
                                                 </select>
 											</div>
-											<div class="form-group col-md-6">
+											<div class="form-group col-md-6 hide " >
 												<label><?php echo $this->lang->line('labels')['selectBranch'];?>*</label>
-												<select name="branch_id" id="branch_id" class=" form-control textinputfields" style="width: 77%">
+												<div id="branches-list">
+												<select name="branch_idss" id="branch_id" class="form-control allowalphanumeric" style="width: 77%">
 												<option value="">Please Select</option>
                                                 </select>
+												
+												</div>
+											</div>
+											<div class="form-group col-md-6 " >
+												<label><?php echo $this->lang->line('labels')['selectBranch'];?>*</label>
+												<select name="branch_id" class="branch_id form-control allowalphanumeric" style="width: 77%" >
+												<option value="">Please Select</option>
+												<?php 
+												foreach($Branches as $br){
+												?>
+												<option value="<?php echo $br->id; ?>"><?php echo $br->text; ?></option>
+												<?php } ?>
+												</select>
 											</div>
 											
-										</div>
-										<div class="col-md-12">
-											<div class="form-group col-md-6">
+											<div class="form-group col-md-6 hide ">
 												<label><?php echo $this->lang->line('labels')['selectDepartment'];?>*</label>
-												<select name="department_id" id="department_id" class=" form-control textinputfields" style="width: 77%">
+												<select name="department_idssss" id="department_id" class="  form-control allowalphanumeric" style="width: 77%">
 												<option value="">Please Select</option>
                                                 </select>
+												
 											</div>
 											
-											<div class="form-group col-md-6">
+											<div class="form-group col-md-6 ">
+												<label><?php echo $this->lang->line('labels')['selectDepartment'];?>*</label>
+												<div class="ddepartment-list">
+												<select name="department_id"  class="DepartmentID form-control allowalphanumeric department_id" style="width: 77%">
+												<option value="">Please Select</option>
+                                                </select>
+												</div>
+											</div>
+											
+										</div>
+										<div class="form-group col-md-6 hide">
                                                <label><?php echo $this->lang->line('labels')['selectDoctor'];?></label>
-                                                <select name="doc_id" id="doc_id" class=" form-control" style="width: 100%">
+                                                <select name="doc_idsss" id="doc_id" class=" form-control" style="width: 100%">
 												<option value="">Please Select</option>
                                                 </select>
+												
                                             </div>
+										<div class="col-md-12">
+											<label><?php echo $this->lang->line('labels')['selectDoctor'];?></label>
+											<div class="doctors-list">
 											
+											</div>
 										</div>
 										<div class="col-md-12">
 											<div class="form-group col-md-6">
@@ -690,8 +718,61 @@ $this->load->view("template/footer.php");
 				        });
 				    }
 				});
+				
+/*$('.getbranches').on('click',function(){
+	 $HospitalID = '<?php echo $this->session->userdata('hospital_id'); ?>';
+	 $.ajax({
+				url: "<?php echo site_url(); ?>/index/searchBranches/",
+				type: "GET",
+				data: { "hospital_id":$HospitalID,"f":"branch_name"},
+				success: function(results) {
+				$("#branches-list").show();
+				$("#branches-list").html(results);
+					
+				},
+				error: function() {
+					callback();
+				}
+			})
+  });*/
+$('.branch_id').on('change',function(){
+   $HospitalID = '<?php echo $this->session->userdata('hospital_id'); ?>';
+   $branchid = $(this).val();
+   //$(".department_id").attr("disabled",false);
+   $.ajax({
+				url: "<?php echo site_url(); ?>index/searchDepartment/",
+				type: "GET",
+				data: { "hospital_id":$HospitalID,"branch_id":$branchid,"f":"department_name"},
+				success: function(results) {
+				$(".ddepartment-list").html(results);
 					
+				},
+				error: function() {
+					callback();
+				}
+			})
+  
+  });
+    
+$(document).on('change','.DepartmentID',function(){
+
+   $departmentid = $(this).val();
+   //$(".department_id").attr("disabled",false);
+   $.ajax({
+				url: "<?php echo site_url(); ?>index/searchDepartmentDoctor/",
+				type: "GET",
+				data: {"dept_id":$departmentid},
+				success: function(results) {
+				$(".doctors-list").html(results);
+					
+				},
+				error: function() {
+					callback();
+				}
+			})
+
+});  	
 
-			});
+});
 
 		</script>
\ No newline at end of file
diff --git a/application/views/Wards/index.php b/application/views/Wards/index.php
index 2e8d8d9..a1ea1d3 100644
--- a/application/views/Wards/index.php
+++ b/application/views/Wards/index.php
@@ -54,7 +54,7 @@ $this->load->view("template/left.php");
 								<div class="">
 									<table id="wards" class="table table-striped table-bordered table-hover table-checkable order-column valign-middle">
 										<thead>
-											<tr><th style="width:10px"></th><th><?php echo $this->lang->line('tableHeaders')['ward'];?></th><th><?php echo $this->lang->line('tableHeaders')['totalBeds'];?></th><th><?php echo $this->lang->line('tableHeaders')['availableBeds'];?></th><th width="20px">#</th>
+											<tr><th style="width:10px"></th><th><?php echo $this->lang->line('tableHeaders')['ward'];?></th><th><?php echo $this->lang->line('tableHeaders')['totalBeds'];?></th><th><?php echo $this->lang->line('tableHeaders')['availableBeds'];?></th><th><?php echo $this->lang->line('tableHeaders')['bed'];?></th><th width="20px">#</th>
 											</tr>
 										</thead>
 										
diff --git a/application/views/template/left.php b/application/views/template/left.php
index 283b997..e40230b 100644
--- a/application/views/template/left.php
+++ b/application/views/template/left.php
@@ -34,8 +34,8 @@
                                 case $this->auth->getAdminRoleType():
                                 //Super Admin
                             ?>
-                                
-                                <li id="li2" class="nav-item">
+                                <li id="li14" class="nav-item"><a href="<?php echo site_url();?>hospitals" class="nav-link"><i class="fa  fa-hospital-o "></i><span class="title"><?php echo $menu['main_hospitals'];?></span></a></li>
+                               <?php /*?> <li id="li2" class="nav-item">
                                     <a href="#" class="nav-link nav-toggle">
                                         <i class="fa fa-hospital-o"></i>
                                         <span><?php echo $menu['main_hospitals'];?></span><span class="arrow"></span>
@@ -47,16 +47,17 @@
                                         <li id="li206" class="nav-item"><a class="nav-link " href="<?php echo site_url();?>wards" ><?php echo $menu['main_wards'];?></a></li> 
                                         <li id="li204" class="nav-item"><a class="nav-link " href="<?php echo site_url();?>beds" ><?php echo $menu['main_beds'];?></a></li>
                                         <li id="li205" class="nav-item"><a class="nav-link " href="<?php echo site_url();?>charges" ><?php echo $menu['main_charges'];?></a></li> 
-                                    </ul>
-                                </li>
+                                    </ul><?php 
+                                </li>*/?>
                                 <li id="li14" class="nav-item"><a href="<?php echo site_url();?>hospital_admin" class="nav-link"><i class="fa  fa-users "></i><span class="title"><?php echo $menu['main_hospital_admin'];?></span></a></li> 
                                 <li id="li4" class="nav-item"><a href="<?php echo site_url();?>doctors" class="nav-link"><i class="menu-icon fa  fa-stethoscope "></i><span class="title"><?php echo $menu['main_dectors'];?> </span></a></li>
                                 <li id="li5" class="nav-item"><a href="<?php echo site_url();?>nurse" class="nav-link"><i class="menu-icon  icon-user "></i><span class="title"><?php echo $menu['main_nurses'];?></span></a></li>
                                 <li id="li6" class="nav-item"><a href="<?php echo site_url();?>receptionist" class="nav-link"><i class="menu-icon  icon-user-following "></i><span class="title"><?php echo $menu['main_receptionists'];?></span></a></li>
-                                <li id="li7" class="nav-item"><a href="<?php echo site_url();?>patients" class="nav-link"><i class="menu-icon icon-users "></i><span class="title"><?php echo $menu['mypulseusers'];?></span></a></li>
-                                <li id="li53" class="nav-item"><a href="<?php echo site_url();?>inpatient" class="nav-link"><i class="menu-icon fa fa-eye "></i><span class="title"><?php echo $menu['main_inpatient_history'];?></span></a></li>
-                                <li id="li8" class="nav-item"><a href="<?php echo site_url();?>medical_store" class="nav-link"><i class="menu-icon fa fa-medkit "></i><span class="title"><?php echo $menu['main_medical_stores'];?></span></a></li>
+								<li id="li8" class="nav-item"><a href="<?php echo site_url();?>medical_store" class="nav-link"><i class="menu-icon fa fa-medkit "></i><span class="title"><?php echo $menu['main_medical_stores'];?></span></a></li>
                                 <li id="li9" class="nav-item"><a href="<?php echo site_url();?>medical_lab" class="nav-link"><i class="menu-icon fa  fa-plus-square "></i><span class="title"><?php echo $menu['main_medical_labs'];?></span></a></li>
+                                <li id="li7" class="nav-item"><a href="<?php echo site_url();?>patients" class="nav-link"><i class="menu-icon icon-users "></i><span class="title"><?php echo $menu['mypulseusers'];?></span></a></li>
+                                <li id="li53" class="nav-item"><a href="<?php echo site_url();?>inpatient" class="nav-link"><i class="menu-icon fa fa-eye "></i><span class="title"><?php echo $menu['main_inpatient'];?></span></a></li>
+                                
                                 <li id="li52" class="nav-item"><a href="<?php echo site_url();?>appoitments" class="nav-link"><i class="menu-icon glyphicon glyphicon-list-alt"></i><span class="title"><?php echo $menu['appoitments'];?></span></a></li>
                                 <?php /*?><li id="li11" class="nav-item"><a href="<?php echo site_url();?>" class="nav-link"><i class="menu-icon fa fa-credit-card "></i><span class="title"><?php echo $menu['main_payments'];?></span></a></li><?php */?>
 
@@ -93,7 +94,7 @@
                                     <a href="<?php echo site_url();?>branches" class="nav-link"><i class=" glyphicon glyphicon-plus-sign"></i><span class="title"><?php echo $menu['main_branches'];?></span>
                                     </a>
                                 </li>
-                                <li class="nav-item" id="li3">
+                                <!--<li class="nav-item" id="li3">
                                     <a href="<?php echo site_url();?>departments" class="nav-link"><i class=" glyphicon glyphicon-plus-sign"></i><span class="title"><?php echo $menu['main_departments'];?></span>
                                     </a>
                                 </li>
@@ -104,7 +105,7 @@
                                 <li class="nav-item" id="li18">
                                     <a href="<?php echo site_url();?>wards" class="nav-link"><i class=" glyphicon glyphicon-plus-sign"></i><span class="title"><?php echo $menu['main_wards'];?></span>
                                     </a>
-                                </li> 
+                                </li> -->
                                 <li class="nav-item" id="li4">
                                     <a href="<?php echo site_url();?>doctors" class="nav-link"><i class=" fa  fa-stethoscope "></i><span class="title"><?php echo $menu['main_dectors'];?> </span>
                                     </a>
@@ -187,6 +188,14 @@
                                 <li class="nav-item" id="li56">
                                     <a href="<?php echo site_url();?>patients/hospital" class="nav-link"><i class="  fa fa-hospital-o "></i><span class="title"><?php echo $menu['main_hospitals'];?></span>
                                     </a>
+                                </li>
+								<li id="li4" class="nav-item">
+								<a href="javascript:void(0);" class="nav-link"><i class="menu-icon fa  fa-stethoscope "></i><span class="title"><?php echo $menu['main_dectors'];?> </span>
+								</a>
+								</li>
+								<li class="nav-item" id="li52">
+                                    <a href="<?php echo site_url();?>appoitments" class="nav-link"><i class="  glyphicon glyphicon-list-alt "></i><span class="title"><?php echo $menu['appoitments'];?></span>
+                                    </a>
                                 </li>
                                 <li class="nav-item" id="li57">
                                     <a href="<?php echo site_url();?>patients/medicalstore" class="nav-link"><i class="fa  fa-users "></i><span class="title"><?php echo $menu['main_medical_stores'];?></span>
@@ -196,11 +205,30 @@
                                     <a href="<?php echo site_url();?>patients/medicallab" class="nav-link"><i class="fa  fa-users "></i><span class="title"><?php echo $menu['main_medical_labs'];?></span>
                                     </a>
                                 </li>
-                                <li class="nav-item" id="li52">
-                                    <a href="<?php echo site_url();?>appoitments" class="nav-link"><i class="  glyphicon glyphicon-list-alt "></i><span class="title"><?php echo $menu['appoitments'];?></span>
+								
+								
+                                <li id="li2" class="nav-item">
+                                    <a href="#" class="nav-link nav-toggle">
+                                        <i class="fa fa-hospital-o"></i>
+                                        <span><?php echo 'Health Records';?></span><span class="arrow"></span>
                                     </a>
+                                    <ul class="sub-menu">
+                                        <li id="li201" class="nav-item">
+										<a class="nav-link " href="javascript:void(0);"><?php echo 'Prescription';?>
+										</a>
+										</li>
+										<li id="li201" class="nav-item">
+										<a class="nav-link " href="javascript:void(0);"><?php echo 'Prognosis';?>
+										</a>
+										</li>
+										<li id="li201" class="nav-item">
+										<a class="nav-link " href="javascript:void(0);"><?php echo 'Reports';?>
+										</a>
+										</li>
+                                    </ul>
                                 </li>
-                                <li class="nav-item" id="li55">
+								
+								<li class="nav-item" id="li55">
                                     <a href="<?php echo site_url();?>patients/inpatient" class="nav-link"><i class="  fa fa-eye"></i><span class="title"><?php echo $menu['main_inpatient_history'];?></span>
                                     </a>
                                 </li>
@@ -232,22 +260,31 @@
                                 break;
                                 case $this->auth->getNurseRoleType():
                             ?>
-                                <li class="nav-item" id="li3">
-                                    <a href="<?php echo site_url();?>nurse/department" class="nav-link"><i class="  fa  fa-users "></i><span class="title"><?php echo $menu['main_departments'];?></span>
-                                    </a>
-                                </li>   
-                                 <li class="nav-item" id="li4">
+                                <li class="nav-item" id="li4">
                                     <a href="<?php echo site_url();?>doctors" class="nav-link"><i class="  fa  fa-stethoscope "></i><span class="title"><?php echo $menu['main_dectors'];?></span>
                                     </a>
                                 </li>
-                                <li class="nav-item" id="li5">
-                                    <a href="<?php echo site_url();?>nurse/beds" class="nav-link"><i class="  fa fa-bed"></i><span class="title"><?php echo $menu['main_beds'];?></span>
+								
+								<li class="nav-item" id="li6">
+                                    <a href="<?php echo site_url();?>nurse/inpatient" class="nav-link"><i class="  fa fa-users"></i><span class="title"><?php echo $menu['main_patients'];?></span>
                                     </a>
                                 </li>
-                                <li class="nav-item" id="li6">
-                                    <a href="<?php echo site_url();?>nurse/inpatient" class="nav-link"><i class="  fa fa-users"></i><span class="title"><?php echo $menu['main_patients'];?></span>
+								
+								<li class="nav-item" id="li3">
+                                    <a href="<?php echo site_url();?>nurse/wards" class="nav-link"><i class="  fa  fa-users "></i><span class="title"><?php echo $menu['main_wards'];?></span>
                                     </a>
                                 </li>
+								
+								<!--<li class="nav-item" id="li3">
+                                    <a href="<?php echo site_url();?>nurse/department" class="nav-link"><i class="  fa  fa-users "></i><span class="title"><?php echo $menu['main_departments'];?></span>
+                                    </a>
+                                </li>-->   
+                                 
+                                <li class="nav-item" id="li5">
+                                    <a href="<?php echo site_url();?>nurse/beds" class="nav-link"><i class="  fa fa-bed"></i><span class="title"><?php echo $menu['main_beds'];?></span>
+                                    </a>
+                                </li>
+                                
                             <?php
                                     
                                 break;
@@ -293,6 +330,11 @@
                             ?>
                                 
                                 <li class="nav-item" id="li81">
+                                    <a href="<?php echo site_url();?>medical_store/orders" class="nav-link"><i class=" glyphicon glyphicon-list-alt"></i><span class="title"><?php echo $menu['orders'];?></span>
+                                    </a>
+                                </li>
+								
+								<li class="nav-item" id="li81">
                                     <a href="<?php echo site_url();?>medical_lab/reports" class="nav-link"><i class=" glyphicon glyphicon-list-alt"></i><span class="title"><?php echo $menu['reports'];?></span>
                                     </a>
                                 </li>
diff --git a/public/assets/js/custom.js b/public/assets/js/custom.js
index 0786760..3203e4c 100644
--- a/public/assets/js/custom.js
+++ b/public/assets/js/custom.js
@@ -30,7 +30,7 @@ $( ".textinputfields" ).keypress(function(event) {
                  return;
         }			
         // allow letters and whitespaces only.
-        if(!(inputValue >= 65 && inputValue <= 120) && (inputValue != 32 && inputValue != 0)) { 
+        if(!(inputValue >= 65 && inputValue <= 122) && (inputValue != 32 && inputValue != 0)) { 
             event.preventDefault(); 
         }
        });
diff --git a/public/tmp/Appoitments.csv b/public/tmp/Appoitments.csv
index a4d97b5..f19629b 100644
--- a/public/tmp/Appoitments.csv
+++ b/public/tmp/Appoitments.csv
@@ -1,6 +1,6 @@
-,,,,,,,,
-5,"<a href='#' data-id='5' class='editbtn' data-toggle='modal' data-target='#edit' data-toggle='tooltip' title='Edit'>APT5</a>","Manoj Vyajpai","Purus Corporation - WB - Dept-1","Dr. Ravi patel",2020-12-10,"00:00:00 to 00:00:00",Rejected,"<span style='display:inline-flex'><a href=""#"" id=""apprlink_5"" class=""apprbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""5"" data-toggle=""tooltip"" data-msg='Do you want to approve the appointment?' title=""Approve"" style='color:green;margin-left:10px'><i class=""glyphicon glyphicon-ok""></i></button></span>"
-6,"<a href='#' data-id='6' class='editbtn' data-toggle='modal' data-target='#edit' data-toggle='tooltip' title='Edit'>APT6</a>","Patient patient","Purus Corporation - WB - Dept-1","Dr. Ravi patel",2021-12-10,"00:00:00 to 00:00:00",Pending,"<span style='display:inline-flex'><a href=""#"" id=""dellink_6"" class=""delbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""6"" data-toggle=""tooltip"" data-msg='Do you want to reject the appointment?' title=""Reject"" style='color:red'><i class=""glyphicon glyphicon-remove""></i></button><a href=""#"" id=""apprlink_6"" class=""apprbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""6"" data-toggle=""tooltip"" data-msg='Do you want to approve the appointment?' title=""Approve"" style='color:green;margin-left:10px'><i class=""glyphicon glyphicon-ok""></i></button></span>"
-7,"<a href='#' data-id='7' class='editbtn' data-toggle='modal' data-target='#edit' data-toggle='tooltip' title='Edit'>APT7</a>","Door Ship","Purus Corporation - WB - Dept-1","Dr. Ravi patel",2017-11-15,"09:00:00 to 09:30:00",Pending,"<span style='display:inline-flex'><a href=""#"" id=""dellink_7"" class=""delbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""7"" data-toggle=""tooltip"" data-msg='Do you want to reject the appointment?' title=""Reject"" style='color:red'><i class=""glyphicon glyphicon-remove""></i></button><a href=""#"" id=""apprlink_7"" class=""apprbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""7"" data-toggle=""tooltip"" data-msg='Do you want to approve the appointment?' title=""Approve"" style='color:green;margin-left:10px'><i class=""glyphicon glyphicon-ok""></i></button></span>"
-8,"<a href='#' data-id='8' class='editbtn' data-toggle='modal' data-target='#edit' data-toggle='tooltip' title='Edit'>APT8</a>","Patient patient","Purus Corporation - WB - Dept-1","Dr. Ravi patel",2017-09-23,"09:00:00 to 09:30:00",Pending,"<span style='display:inline-flex'><a href=""#"" id=""dellink_8"" class=""delbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""8"" data-toggle=""tooltip"" data-msg='Do you want to reject the appointment?' title=""Reject"" style='color:red'><i class=""glyphicon glyphicon-remove""></i></button><a href=""#"" id=""apprlink_8"" class=""apprbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""8"" data-toggle=""tooltip"" data-msg='Do you want to approve the appointment?' title=""Approve"" style='color:green;margin-left:10px'><i class=""glyphicon glyphicon-ok""></i></button></span>"
-12,"<a href='#' data-id='12' class='editbtn' data-toggle='modal' data-target='#edit' data-toggle='tooltip' title='Edit'>APT12</a>",,"Purus Corporation - WB - Dept-1","Dr. Ravi patel",2018-05-25,"09:00:00 to 09:30:00",Approved,"<span style='display:inline-flex'>-</span>"
+,,,,,,,
+5,"<a href='http://localhost:8080/mypulse//doctors/patientRecord/5' >APT5</a>","Manoj Vyajpai",asdasa,2020-12-10,"00:00:00 to 00:00:00",Rejected,"<span style='display:inline-flex'><a href=""#"" id=""apprlink_5"" class=""apprbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""5"" data-toggle=""tooltip"" data-msg='Do you want to approve the appointment?' title=""Approve"" style='color:green;margin-left:10px'><i class=""glyphicon glyphicon-ok""></i></button><a href='#' data-id='5' class='editbtn' data-toggle='modal' data-target='#edit' data-toggle='tooltip' title='Edit' style='margin-left:10px'><i class='fa fa-pencil'></i></a></span>"
+6,"<a href='http://localhost:8080/mypulse//doctors/patientRecord/6' >APT6</a>","Patient patient",K,2021-12-10,"00:00:00 to 00:00:00",Pending,"<span style='display:inline-flex'><a href=""#"" id=""dellink_6"" class=""delbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""6"" data-toggle=""tooltip"" data-msg='Do you want to reject the appointment?' title=""Reject"" style='color:red'><i class=""glyphicon glyphicon-remove""></i></button><a href=""#"" id=""apprlink_6"" class=""apprbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""6"" data-toggle=""tooltip"" data-msg='Do you want to approve the appointment?' title=""Approve"" style='color:green;margin-left:10px'><i class=""glyphicon glyphicon-ok""></i></button><a href='#' data-id='6' class='editbtn' data-toggle='modal' data-target='#edit' data-toggle='tooltip' title='Edit' style='margin-left:10px'><i class='fa fa-pencil'></i></a></span>"
+7,"<a href='http://localhost:8080/mypulse//doctors/patientRecord/7' >APT7</a>","Door Ship",asdf,2017-11-15,"09:00:00 to 09:30:00",Pending,"<span style='display:inline-flex'><a href=""#"" id=""dellink_7"" class=""delbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""7"" data-toggle=""tooltip"" data-msg='Do you want to reject the appointment?' title=""Reject"" style='color:red'><i class=""glyphicon glyphicon-remove""></i></button><a href=""#"" id=""apprlink_7"" class=""apprbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""7"" data-toggle=""tooltip"" data-msg='Do you want to approve the appointment?' title=""Approve"" style='color:green;margin-left:10px'><i class=""glyphicon glyphicon-ok""></i></button><a href='#' data-id='7' class='editbtn' data-toggle='modal' data-target='#edit' data-toggle='tooltip' title='Edit' style='margin-left:10px'><i class='fa fa-pencil'></i></a></span>"
+8,"<a href='http://localhost:8080/mypulse//doctors/patientRecord/8' >APT8</a>","Patient patient",ASD,2017-09-23,"09:00:00 to 09:30:00",Pending,"<span style='display:inline-flex'><a href=""#"" id=""dellink_8"" class=""delbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""8"" data-toggle=""tooltip"" data-msg='Do you want to reject the appointment?' title=""Reject"" style='color:red'><i class=""glyphicon glyphicon-remove""></i></button><a href=""#"" id=""apprlink_8"" class=""apprbtn""  data-toggle=""modal"" data-target="".bs-example-modal-sm"" data-id=""8"" data-toggle=""tooltip"" data-msg='Do you want to approve the appointment?' title=""Approve"" style='color:green;margin-left:10px'><i class=""glyphicon glyphicon-ok""></i></button><a href='#' data-id='8' class='editbtn' data-toggle='modal' data-target='#edit' data-toggle='tooltip' title='Edit' style='margin-left:10px'><i class='fa fa-pencil'></i></a></span>"
+12,"<a href='http://localhost:8080/mypulse//doctors/patientRecord/12' >APT12</a>",,"general checkup",2018-05-25,"09:00:00 to 09:30:00",Approved,"<span style='display:inline-flex'><a href='#' data-id='12' class='editbtn' data-toggle='modal' data-target='#edit' data-toggle='tooltip' title='Edit' style='margin-left:10px'><i class='fa fa-pencil'></i></a></span>"
diff --git a/public/tmp/Receptionist.csv b/public/tmp/Receptionist.csv
index 8215541..5860132 100644
--- a/public/tmp/Receptionist.csv
+++ b/public/tmp/Receptionist.csv
@@ -1,5 +1,7 @@
 #,Receptionist,Hospital,Branch,Department,Doctor,Status
-1,"Receptionist Patel",Hospital1,h1_Branch1,h1_b1_Dept2,"Dr. Yogesh  Patel",Active
-2,"rece Patel",Hospital1,h1_Branch1,h1_b1_Dept2,"Dr. Yogesh  Patel",Active
-4,,Hospital1,h1_Branch1,h1_b1_Dept2,"Dr. Yogesh  Patel",Active
-5,,Hospital1,h1_Branch1,h1_b1_Dept2,"Dr. Yogesh  Patel",Active
+1,"Raju patel","Purus Corporation",WB,Dept-1,"Dr. Ravi patel",Active
+2,"Door Ship","Purus Corporation",WB,Dept-1,"Dr. Ravi patel",Active
+3,"Receptionist ","Purus Corporation",WB,Dept-1,"Dr. Ravi patel",Active
+4,"Yogesh Patel","Purus Corporation",WB,Dept-1,"Dr. Ravi patel",Active
+5,"receptionist receptionist","Purus Corporation",WB,Dept-1,"Dr. Ravi patel",Active
+13,"ravi kumar ravi kumar","Purus Corporation",WB,Dept-1,"Dr. Ravi patel",Active
